src/test/java/org/prebid/server/bidder/openx/OpenxBidderTest.java

```diff
@@ -0,0 +1,467 @@
+                                .id("impId1")
+                                .banner(Banner.builder().build())
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<BidderBid>> result = openxBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .containsOnly(BidderBid.of(
+                        Bid.builder()
+                                .impid("impId1")
+                                .price(BigDecimal.ONE)
+                                .dealid("dealid")
+                                .w(200)
+                                .h(150)
+                                .adm("<div>This is an Ad</div>")
+                                .build(),
+                        BidType.banner));
+    }
+
+    @Test
+    public void makeBidsShouldReturnRespectBannerImpWhenBothBannerAndVideoImpWithSameIdExist() throws JsonProcessingException {
+        // given
+        final HttpCall httpCall = givenHttpCall(200, mapper.writeValueAsString(BidResponse.builder()
+                .seatbid(singletonList(SeatBid.builder()
+                        .bid(singletonList(Bid.builder()
+                                .w(200)
+                                .h(150)
+                                .price(BigDecimal.ONE)
+                                .impid("impId1")
+                                .dealid("dealid")
+                                .adm("<div>This is an Ad</div>")
+                                .build()))
+                        .build()))
+                .build()));
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .id("bidRequestId")
+                .imp(asList(
+                        Imp.builder()
+                                .id("impId1")
+                                .video(Video.builder().build())
+                                .banner(Banner.builder().build())
+                    .build()))
+                .build();
+
+        // when
+        final Result<List<BidderBid>> result = openxBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .containsOnly(BidderBid.of(
+                        Bid.builder()
+                                .impid("impId1")
+                                .price(BigDecimal.ONE)
+                                .dealid("dealid")
+                                .w(200)
+                                .h(150)
+                                .adm("<div>This is an Ad</div>")
+                                .build(),
+                        BidType.banner));
+    }
+
+    private static HttpCall givenHttpCall(int statusCode, String body) {
+        return HttpCall.full(null, HttpResponse.of(statusCode, null, body), null);
+    }
+
```

Comment:
Maybe some of below tests can be useful? 
- makeHttpRequest when Imp has (Banner or Video) + Audio type at the same time.
- makeHttpRequest when valid request + error at the same time was returned.
- makeBids when bidResponse is null or getSeatBid return null
- extractTargeting test

Reply:
First two items covered in updated makeHttpRequestsShouldReturnResultWithExpectedFieldsSet;
For the third item makeBidsShouldReturnResultContainingEmptyValueAndErrorsWhenSeatBidEmpty was added;
And the last one also fulfilled.